dbfit-java/core/src/main/java/dbfit/api/DbStoredProcedureCall.java

```diff
@@ -51,7 +51,12 @@ void bindParametersTo(StatementExecution cs) throws SQLException {
     }
 
     public StatementExecution toStatementExecution() throws SQLException {
-        StatementExecution cs = new StatementExecution(this.environment.getConnection().prepareCall(toSqlString()));
+        StatementExecution cs;
+        if (isFunction()) {
+            cs = environment.createFunctionStatementExecution(this.environment.getConnection().prepareCall(toSqlString()), true);
+        } else {
+            cs = environment.createStatementExecution(this.environment.getConnection().prepareCall(toSqlString()), true);
```

Comment:
I'll get rid of that.

I was also going to update this snippet as per the comment previously posted (starting with): -

> Do you mean some refactoring or re-organsing what's already there?